src/main/java/com/google/sps/servlets/GetBlurAreasServlet.java

```diff
@@ -0,0 +1,186 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/get-blur-areas")
+public class GetBlurAreasServlet extends HttpServlet {
+
+  /**
+   * This method handles the POST requests to "/get-blur-areas". Receives a BlobKey parameter which
+   * represents the Blobstore location of an image. Responds with a JSON ArrayList of polygons. A
+   * polygon is represented by a List of points. A point contains two properties, its coordinates: x
+   * and y.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the BlobKey that points to the image uploaded by the user.
+    BlobKey blobKey = getBlobKey(request, "image");
+
+    // User didn't upload a file, so render an error message.
+    if (blobKey == null) {
+      response.getWriter().println("Please upload an image file.");
+      return;
+    }
+
+    // Get the image the user uploaded as bytes.
+    byte[] imageBytes = getBlobBytes(blobKey);
+
+    ArrayList<List<Point>> blurAreas = getBlurAreas(imageBytes);
+
+    // Delete the image from the blobstore.
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    blobstoreService.delete(blobKey);
+
+    // Convert the polygons to JSON.
+    Gson gson = new Gson();
+    String jsonResponse = gson.toJson(blurAreas);
+
+    // Send the JSON back as the response.
+    response.setContentType("application/json");
+    response.getWriter().println(jsonResponse);
+  }
+
+  /**
+   * Uses the Google Cloud Vision API to find faces in the image represented by the binary data
+   * stored in @param imageBytes
+   *
+   * @return an ArrayList of bounding polygons representing faces. Polygons are represented by a
+   *     list of points.
+   */
+  private ArrayList<List<Point>> getBlurAreas(byte[] imageBytes) throws IOException {
+    // This is the array that we will return.
+    ArrayList<List<Point>> facePolys = new ArrayList<>();
+
+    // Convert bytes to an Image object.
+    ByteString byteString = ByteString.copyFrom(imageBytes);
+    Image image = Image.newBuilder().setContent(byteString).build();
+
+    // Create a face detection request to our image.
+    Feature feature = Feature.newBuilder().setType(Feature.Type.FACE_DETECTION).build();
+    AnnotateImageRequest request =
+        AnnotateImageRequest.newBuilder().addFeatures(feature).setImage(image).build();
+
+    // Add request to an array.
+    List<AnnotateImageRequest> requests = new ArrayList<>();
+    requests.add(request);
+
+    // Annotate the image from our request.
+    try (ImageAnnotatorClient client = ImageAnnotatorClient.create()) {
+      BatchAnnotateImagesResponse response = client.batchAnnotateImages(requests);
+      List<AnnotateImageResponse> responses = response.getResponsesList();
+
+      for (AnnotateImageResponse res : responses) {
+        if (res.hasError()) {
+          continue;
+        }
```

Comment:
Is this an expected error? What is the reason why it could happen? Don't you want to send it to the client probably to inform the user about the error?

Reply:
A comment in the code will help. Please expand a bit on what kind of errors are expected so that we are simply skipping them.

Reply:
I added a comment and logged the error in the default console.